### PR TITLE
Support install/uninstall addon via rosa cli

### PR DESCRIPTION
**Pull request by oharan2**
Accept `rosa` attribute to indicate the ROSA cli direct usage for addon installation.

### CODE REVIEW

File: ocm_python_wrapper/cluster.py [hunk 1]
```diff
@@ -257,16 +258,27 @@ def addon_info(self):
...
 
-    def validate_addon_parameters(self, parameters):
-        _info = self.addon_info()
-        _parameters = _info.get("parameters")
-        if not _parameters and parameters:
+    def validate_addon_parameters(self, user_parameters):
+        """Checks user input for addon parameters.
+
+        Args:
+            user_parameters (List) : User parameter input set with values.
```
Comment (rnetser) [inline ocm_python_wrapper/cluster.py:265]: `User parameter input set with values.` please rephrase; what does `set with values` mean?

File: ocm_python_wrapper/cluster.py [hunk 2]
```diff
@@ -257,16 +258,27 @@ def addon_info(self):
...
+
+        Returns: Updated parameters (if default values updated) to provide for installation.
```
Comment (rnetser) [inline ocm_python_wrapper/cluster.py:267]: missing newline

Comment (rnetser) [reply ocm_python_wrapper/cluster.py:267]: the function does not return anything

File: ocm_python_wrapper/cluster.py [hunk 3]
```diff
@@ -303,12 +322,24 @@ def install_addon(self, parameters=None, wait=True, wait_timeout=TIMEOUT_30MIN):
...
 
         LOGGER.info(f"Installing addon {self.addon_name} v{self.addon_version}")
-        res = self.client.api_clusters_mgmt_v1_clusters_cluster_id_addons_post(
-            cluster_id=self.cluster_id,
-            add_on_installation=AddOnInstallation(
-                _check_type=False, **_addon_installation_dict
-            ),
-        )
+        if rosa_cli:
+            # Billing-model is relevant for installation
```
Comment (rnetser) [inline ocm_python_wrapper/cluster.py:326]: is it relevant to all addons installed via cli?

File: pyproject.toml [hunk 4]
```diff
@@ -41,6 +41,8 @@ dependencies = [
     "requests",
     "pyyaml",
     "openshift-python-utilities",
+    "python-benedict",
```
Comment (rnetser) [inline pyproject.toml:44]: not part of this pr

**Review by rnetser — CHANGES_REQUESTED** [review]

File: ocm_python_wrapper/cluster.py [hunk 5]
```diff
@@ -278,23 +290,30 @@ def validate_addon_parameters(self, parameters):
                 f"{self.addon_name} missing some required parameters {missing_parameter}"
             )
 
-    def install_addon(self, parameters=None, wait=True, wait_timeout=TIMEOUT_30MIN):
+    def install_addon(
+        self,
+        parameters=None,
+        wait=True,
+        wait_timeout=TIMEOUT_30MIN,
+        rosa_cli=False,
```
Comment (omrirh) [inline ocm_python_wrapper/cluster.py:298]:
```suggestion
        rosa=False,
```

`run_action` passes `rosa` param to `install_addon` kwargs as shown [here](https://github.com/RedHatQE/ocp-addons-operators-cli/blob/85a0cbe9c9f054bed19b0f80ac51aa8632a5ed21/app/addon_cmds/commands.py#L16)

File: pyproject.toml [hunk 6]
```diff
@@ -41,6 +41,7 @@ dependencies = [
     "requests",
     "pyyaml",
     "openshift-python-utilities",
+    "rosa-pyhon-client"
```
Comment (myakove) [inline pyproject.toml:44]: typo

**Review by myakove — CHANGES_REQUESTED** [review]

**Review by rnetser — APPROVED** [review]

**Review by myakove — APPROVED** [review]

**Pull request merged**